File: 2022/day01/typescript/solution.ts
export function partOne(input: string) {
    let highTotal = 0;
    let elves = parseElves(input);

    elves.forEach((elf) => {
        const calories = elf.reduce((sum, el) => (sum += Number(el)), 0);

        if (calories > highTotal) highTotal = calories;
    });

    return highTotal;
}

export function partTwo(input: string) {
    let elves = parseElves(input);

    let sortedElves = elves
        .map((elf) => elf.reduce((sum, el) => (sum += Number(el)), 0)) // get each elf total cals
        .sort((a, b) => b - a);

    return sortedElves[0] + sortedElves[1] + sortedElves[2];
}

function parseElves(input: string) {
    return input
        .split(/\n\n+/g) //Split input per elf
        .map((elf) => elf.split(/\n/g)); // map food items per elf
}

File: 2022/day01/python/solution.py
def part_one(input):

    high = 0
    for elf in input.strip().split("\n\n"):
        total = 0
        for food in elf.split("\n"):
            total += int(food)

        if total > high:
            high = total

    return high


def part_two(input):
    elfTotals = []
    for elf in input.strip().split("\n\n"):
        total = 0
        for food in elf.split("\n"):
            total += int(food)

        elfTotals.append(total)

    elfTotals.sort(reverse=True)

    topThreeTotal = 0
    for i in range(0, 3):
        topThreeTotal += elfTotals[i]

    return topThreeTotal


with open("../input", "r") as file:
    input = file.read()
    print("part one:", part_one(input))
    print("part two:", part_two(input))
